packages/astro/src/actions/runtime/virtual/server.ts

```diff
@@ -212,3 +230,80 @@ function unwrapBaseObjectSchema(schema: z.ZodType, unparsedInput: FormData) {
 	}
 	return schema;
 }
+
+/**
+ * Access information about Action requests from middleware.
+ */
+export function getMiddlewareContext(context: APIContext) {
```

Comment:
I would suggest a different name here. The middleware already accepts a context `ctx: APIContext` as a first argument. Having a function named `getMiddlewareContext` is confusing and doesn't provide enough information of what it does. What about `getActionContext`?

Also, since this is a user API now, we should type the return type of the function, and expose it to userland. They may need it.